app/models/update_expense_and_receipt.rb

```diff
@@ -0,0 +1,51 @@
+class UpdateExpenseAndReceipt
+  attr_reader :expense
+
+  def initialize(expense_params)
+    @params = expense_params
+  end
+
+  def update(expense)
+    setup(expense)
+    unless update_expense && update_or_create_receipt
```

Comment:
This should be a transaction. Because you could have `update_expense` changing stuff in DB and then `upgrade_or_create_receipt` could fail and you'd return false, and then trigger error on controller.